drivers/syscon/syscon_bflb_efuse.c

```diff
@@ -0,0 +1,256 @@
+
+	/* do a 'ahb clock' setup */
+	tmp =	EF_CTRL_EFUSE_CTRL_PROTECT
+		| (EF_CTRL_OP_MODE_AUTO << EF_CTRL_EF_IF_0_MANUAL_EN_POS)
+		| (EF_CTRL_PARA_DFT << EF_CTRL_EF_IF_0_CYC_MODIFY_POS)
+#if defined(CONFIG_SOC_SERIES_BL60X)
+		| (EF_CTRL_SAHB_CLK << EF_CTRL_EF_CLK_SAHB_DATA_SEL_POS)
+#endif
+		| (1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS)
+		| (0 << EF_CTRL_EF_IF_POR_DIG_POS)
+		| (1 << EF_CTRL_EF_IF_0_INT_CLR_POS)
+		| (0 << EF_CTRL_EF_IF_0_RW_POS)
+		| (0 << EF_CTRL_EF_IF_0_TRIG_POS);
+
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+	efuse_bflb_clock_settle();
+
+	/* clear PDS cache registry */
+	for (uint32_t i = 0; i < config->size / 4; i++) {
+		pefuse_start[i] = 0;
+	}
+
+	/* Load efuse region0 */
+	/* not ahb clock setup */
+	tmp =	EF_CTRL_EFUSE_CTRL_PROTECT
+		| (EF_CTRL_OP_MODE_AUTO << EF_CTRL_EF_IF_0_MANUAL_EN_POS)
+		| (EF_CTRL_PARA_DFT << EF_CTRL_EF_IF_0_CYC_MODIFY_POS)
+#if defined(CONFIG_SOC_SERIES_BL60X)
+		| (EF_CTRL_EF_CLK << EF_CTRL_EF_CLK_SAHB_DATA_SEL_POS)
+#endif
+		| (1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS)
+		| (0 << EF_CTRL_EF_IF_POR_DIG_POS)
+		| (1 << EF_CTRL_EF_IF_0_INT_CLR_POS)
+		| (0 << EF_CTRL_EF_IF_0_RW_POS)
+		| (0 << EF_CTRL_EF_IF_0_TRIG_POS);
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+
+	/* trigger read */
+	tmp =	EF_CTRL_EFUSE_CTRL_PROTECT
+		| (EF_CTRL_OP_MODE_AUTO << EF_CTRL_EF_IF_0_MANUAL_EN_POS)
+		| (EF_CTRL_PARA_DFT << EF_CTRL_EF_IF_0_CYC_MODIFY_POS)
+#if defined(CONFIG_SOC_SERIES_BL60X)
+		| (EF_CTRL_EF_CLK << EF_CTRL_EF_CLK_SAHB_DATA_SEL_POS)
+#endif
+		| (1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS)
+		| (0 << EF_CTRL_EF_IF_POR_DIG_POS)
+		| (1 << EF_CTRL_EF_IF_0_INT_CLR_POS)
+		| (0 << EF_CTRL_EF_IF_0_RW_POS)
+		| (1 << EF_CTRL_EF_IF_0_TRIG_POS);
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+	efuse_bflb_clock_delay_32M_ms(5);
+
+	/* wait for read to complete */
+	do {
+		efuse_bflb_clock_delay_32M_ms(1);
+		tmp = sys_read32(config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+	} while ((tmp & EF_CTRL_EF_IF_0_BUSY_MSK) ||
+		!(tmp & EF_CTRL_EF_IF_0_AUTOLOAD_DONE_MSK));
+
+	/* do a 'ahb clock' setup */
+	tmp =	EF_CTRL_EFUSE_CTRL_PROTECT
+		| (EF_CTRL_OP_MODE_AUTO << EF_CTRL_EF_IF_0_MANUAL_EN_POS)
+		| (EF_CTRL_PARA_DFT << EF_CTRL_EF_IF_0_CYC_MODIFY_POS)
+#if defined(CONFIG_SOC_SERIES_BL60X)
+		| (EF_CTRL_SAHB_CLK << EF_CTRL_EF_CLK_SAHB_DATA_SEL_POS)
+#endif
+		| (1 << EF_CTRL_EF_IF_AUTO_RD_EN_POS)
+		| (0 << EF_CTRL_EF_IF_POR_DIG_POS)
+		| (1 << EF_CTRL_EF_IF_0_INT_CLR_POS)
+		| (0 << EF_CTRL_EF_IF_0_RW_POS)
+		| (0 << EF_CTRL_EF_IF_0_TRIG_POS);
+
+	sys_write32(tmp, config->addr + EF_CTRL_EF_IF_CTRL_0_OFFSET);
+}
+
+static void efuse_bflb_cache(const struct device *dev)
+{
+	struct efuse_bflb_data *data = dev->data;
+	const struct efuse_bflb_config *config = dev->config;
+	uint32_t tmp = 0;
+	uint8_t old_clock_root;
+
+	tmp = sys_read32(HBN_BASE + HBN_GLB_OFFSET);
+	old_clock_root = (tmp & HBN_ROOT_CLK_SEL_MSK) >> HBN_ROOT_CLK_SEL_POS;
```

Comment:
nit: we can have a matching helper `efuse_bflb_get_root_clock()` function
```suggestion
	old_clock_root = efuse_bflb_get_root_clock();
```

Reply:
See clock driver comment